Changes by ChibiShibe

chibishibe opened the pull request:
Re-instated the following behaviors:
-Greet
-Eventbrite
-countdown

Changes to Greet:
indexed it
Added personality

Changes to eventbrite:
Re-instated Token
Re-instated no peek function
Added heightened personality to no peek

Changes to Clap:
Changed Class name to fit behaviour

Changes to countdown:
Fixed Time format, stopped console error

~~New additions
"Do the impossible" behavior key-phrase response
-Reference to TTGL anime show
"Shake"
-Converts messages sent via "!shake" to Gif emoji varients"


~~Other changes
Swapped icons around, Began centralizing

chibishibe committed:
making changes for API

chibishibe committed:
changed con Date

chibishibe committed:
testing Null return fix

chibishibe committed:
undo test fix for Null

drat

chibishibe committed:
Null roll fix Test 2

Let's try this shall we? 
Duped Section that normally scream when you roll more than 100

chibishibe committed:
Changed Line 53 to specify Less than 1 for input

It kept spitting out "(Please enter number of dice being rolled and Number of Sides like "!rtd 1d5)" for every roll because input.

chibishibe committed:
Dice

1d100 function reply doesn't work
No input still returning null

chibishibe committed:
Revert changes to original ((rtd fix test))

UGH, Reverted to baseline

chibishibe committed:
slight edit to test git pull

yee

chibishibe committed:
change back to finalize git pull test

test 1/1 succeed

chibishibe committed:
No peeking

Test

chibishibe committed:
Changed icon

chibishibe committed:
added character1

Gave a little personality

chibishibe committed:
EVEN MORE CHARACTER

because promises amiright?

chibishibe committed:
added more response

Derp Soon

chibishibe committed:
Removed Peek, Testing EventBrite

Changed keys and tokens. LET'S FIRE

chibishibe committed:
Put lock on

Response back

chibishibe committed:
shortend and hid link 

weee

chibishibe committed:
Greet addition 

Added Greet

chibishibe committed:
Forgot a " ' "

whoops Round 2

chibishibe committed:
removed space

back to planet, remove the space

chibishibe committed:
Space removed

no space here either

chibishibe committed:
comma?

lets try a comma

chibishibe committed:
remove other things

Back to square 1?

chibishibe committed:
fresh?

not a total revision

chibishibe committed:
Square 0

Zero

chibishibe committed:
rd 2 fresh

goddamnitall

chibishibe committed:
rd3

damndamnda

chibishibe committed:
space put back

Turns out we might need it?

chibishibe committed:
rd4

killme

chibishibe committed:
OKAY LETS DO THAT AGAIN

WEEEEEEEEEEEE Greet

chibishibe committed:
personality

the cult of it with greet

chibishibe committed:
TryingSomethingnew

test

chibishibe committed:
import for the impossible

GIGA
DRILLLLLLLLLLLLLLLLLL
BREAAAAKKKERRRRRRRRRRRRRRRRRRRRRRRRRR

chibishibe committed:
changin more

MORE DRILLLLLL

chibishibe committed:
Cool

nothing to see here

chibishibe committed:
personal change to greet descript

Made it a bit less generic

chibishibe committed:
fixed a semi

semi fixed... wait no, it's fully fixed

chibishibe committed:
I gotta know

PLEASE SHOW ME THE GIF

chibishibe committed:
Change back

Test confirmed, Hoof_beatz can use Gif emojis as profile pics and they work

chibishibe committed:
Let's try hiding it

chibishibe committed:
eh

nope

chibishibe committed:
Changed MaxDays+Condate

Yeee

chibishibe committed:
Format time chang+Timezoneadd

LEts do this?

chibishibe committed:
Create shake.js

chibishibe committed:
Test

I hate life

chibishibe committed:
update to test

still hating life

chibishibe committed:
Brackremoval

Bracket

chibishibe committed:
reset

reset

chibishibe committed:
module where?

Damnit Herewego.png

chibishibe committed:
caps maybe

maybe

chibishibe committed:
Emergency change

lol

chibishibe committed:
fullstop

staph

chibishibe committed:
ve -> ce

typo

chibishibe committed:
Changing things

yeeup

chibishibe committed:
Now replace the space with blank

blank

chibishibe committed:
Array

alpha array

chibishibe committed:
ugh

ugh

chibishibe committed:
Array

arararary

chibishibe committed:
array

aararray

chibishibe committed:
aray

ARRAY

chibishibe committed:
test

tsst

chibishibe committed:
changed !help view

!help view change

chibishibe committed:
test

tesboi

chibishibe committed:
gg

everyone

chibishibe committed:
update revision

Test

chibishibe committed:
class name 

update

chibishibe committed:
Clap class name

change

chibishibe committed:
literal template

Solves  template

chibishibe committed:
Bracket

bracketbracketbracketbracketbracketbracketbracket

chibishibe committed:
Limiter

LIMIT THE CODE

chibishibe committed:
Limiter fixed response

asd

chibishibe committed:
typo

typo

chibishibe committed:
added warning

angri

chibishibe committed:
removed warning test

Great on paper assholish in practice

chibishibe committed:
Removed length

length remove to end null?

chibishibe committed:
Update

Reduced

chibishibe committed:
Fixed typo, added personality

No more console spam

chibishibe committed:
intro

intro

chibishibe committed:
introchange

test

chibishibe committed:
index intro

introductions

chibishibe committed:
, -> ;

chibishibe committed:
testfix

sxytax error fix

chibishibe committed:
Update index.js

chibishibe committed:
Update introductions.js

chibishibe committed:
Update index.js

chibishibe committed:
Corrected time

28th->27th

chibishibe committed:
27-26th

Accounted for staff on thursday

chibishibe committed:
Timefixondate

On my hands...too much

chibishibe committed:
Changed cron Format

Specified DAILY and Hour Functions

chibishibe committed:
Added Node for node.schedule

Forgot the astrike

chibishibe committed:
Cron Format Update

Am I a timelord yet?

chibishibe committed:
addional changes

13:00 daily

chibishibe committed:
exp. test

1pm Monday, Sunday(test)

chibishibe committed:
test

test

chibishibe committed:
Update days-until.js

chibishibe committed:
test

test

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update random-channel.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update index.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Change bot.log to console.log

(guarantee that logging goes to muh terminal thx)

chibishibe committed:
Update time

chibishibe committed:
check return of _api

chibishibe committed:
yay

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Test

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update days-until.js

chibishibe committed:
Update random-channel.js

chibishibe committed:
Update random-channel.js

chibishibe committed:
Create RubyRave

chibishibe committed:
added rave

chibishibe committed:
Rename rave to rave.js

chibishibe committed:
Delete RubyRave

chibishibe committed:
Rename src/behaviors/Rave/rave.js to src/behaviors/rave/rave.js

chibishibe committed:
Update index.js

chibishibe committed:
Update rave.js

chibishibe committed:
Update clap.js

imjoshdean commented on package.json:
```diff
@@ -48,6 +48,8 @@
     "rimraf": "^2.5.2"
   },
   "dependencies": {
+    "airtable": "^0.5.2",
```
This isn't being used, please remove.

Grissess replied:
Oh shoot, this PR's off a branch, isn't it...
That's in place pending implementation as a Karma backend (soon).
If you want, we can revert and drop a branch with this deleted at ~that point in time :)

imjoshdean replied:
You can also just throw up another commit removing the dependencies, less work for you.

imjoshdean commented on package.json:
```diff
@@ -48,6 +48,8 @@
   },
   "dependencies": {
+    "airtable": "^0.5.2",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
```
This isn't being used, please remove.

imjoshdean commented on src/behaviors/clap/clap.js:
```diff
@@ -1,28 +1,28 @@
 import Behavior from '../behavior.js';
 
-class RollTheDice extends Behavior {
+class ClaporRave extends Behavior {
```
Capitalize your "o" with CamelCase you capitalize every letter. So `ClapOrRave`

imjoshdean commented on src/behaviors/clap/clap.js:
```diff
@@ -1,28 +1,28 @@
   constructor(settings) {
-    settings.name = 'Clap Hype';
-    settings.description = `'cause 👏 sometimes 👏 you 👏 need 👏 emphasis!`;
+    settings.name = 'ClappingRave';
```
Split these words, the name is more meant to be user readable.

imjoshdean commented on src/behaviors/clap/clap.js:
```diff
@@ -1,28 +1,28 @@
-    });
+    ['clap', 'rubyrave', 'wut'].forEach(emoji => this.commands.push({
+      tag: emoji,
+      description: `I'll hype your message, !Clap or !Rave`
```
Will `!Clap` work, or only `!clap`? You should make sure what users see is what they can use. I've tried to strive for all lower case for these where possible.

chibishibe replied:
Agreed, Pesky Shift Trigger Finger. 

Will lowercase

imjoshdean commented on src/behaviors/clap/clap.js:
```diff
@@ -1,28 +1,28 @@
+    const parsedMessage = this.parseMessage(message, command);
 
-    this.bot.postMessage(channel, `${parsedMessage.join(' :clap: ')} :clap:`, {
+    this.bot.postMessage(channel, `${parsedMessage.join(' :' + command + ': ')} :${command}:`, {
```
Neat thing about template literals is that you can nest them!

```js
`${parsedMessage.join(`:${command}: `)} :${command}:`
```

Grissess replied:
_You can?!_ I was scared ;-;

chibishibe replied:
Woohoo naive guessing worked!

imjoshdean commented on src/behaviors/days-until/days-until.js:
```diff
@@ -101,21 +101,22 @@ class DaysUntil extends Behavior {
       }
 
       topic = topic.replace(/\d+ days? until BronyCon!/gi, message);
+      console.log(topic);
```
Please go back to `bot.log` for these. There is a long term desire to want to explicitly log out to a file via the bot that I would like to implement and keeping the logging to the bot would help with that.

Grissess replied:
We weren't sure it was functioning in those cases; in fact, those were never called anyway. It seems like `Promise.catch` is more reliable in these cases, and was used for quite a bit of the debugging around there. (Turns out you can't set the topic of a channel you're not in--but you can certainly post there! :shrug:)

imjoshdean commented on src/behaviors/eventbrite/eventbrite.js:
```diff
@@ -29,6 +29,14 @@ class EventBrite extends Behavior {
 
   execute(command, message, channel, messageData) {
+
+   this.bot.postMessage(channel, `Sorry guys, I'm good at dropping beatz, not numbers.. I'll look around for you though okay? Here's a picture! https://goo.gl/DWWqs4`, {
```
Extra period after numbers.

chibishibe replied:
Added another period to demonstrate ellipsis...

imjoshdean commented on src/behaviors/greet/greet.js:
```diff
@@ -6,14 +6,14 @@ export default class GreetBehavior extends Behavior {
     super(settings);
 
-    this.addCommand('hello', 'Say hello back whenever someone types !hello');
+    this.addCommand('hello', 'Say !hello and I\'ll say hi back!');
```
Switch to back ticks and you won't need to escape your '

``` `Say !hello and I'll say hi back!` ```

imjoshdean commented on src/behaviors/greet/greet.js:
```diff
@@ -6,14 +6,14 @@ export default class GreetBehavior extends Behavior {
-      this.bot.postMessage(channel, `Hello to you too, <@${data.user}>!`, {
-        icon_emoji: ':wave:'
+      this.bot.postMessage(channel, `Yo, What's goin on! Enjoyin' Slack? <@${data.user}>!`, {
+        icon_emoji: ':Hoof_beatz:'
```
Lower case h

imjoshdean commented on src/behaviors/impossible/impossible.js:
```diff
@@ -0,0 +1,23 @@
+    switch (command) {
+    case 'dotheimpossible':
+      this.bot.postMessage(channel, `<@${data.user}>:sh-r::sh-o::sh-w::sh-r::sh-o::sh-w::blank::sh-f::sh-i::sh-g::sh-h::sh-t::blank::sh-t::sh-h::sh-e::blank::sh-p::sh-o::sh-w::sh-a::sh-h:`, {
+        icon_emoji: ':rowrowHoof_beatz:'
```
Lower case h. This works, but it's just better to keep consistent across the board.

imjoshdean commented on src/behaviors/introductions/introductions.js:
```diff
@@ -12,14 +12,13 @@ class Introductions extends Behavior {
-      const [josh, sheva] = users,
-        joshTag = `<@${josh.id}|${josh.name}>`,
-        shevaTag = `<@${sheva.id}|${sheva.name}>`;
+    Promise.all([bot.getUser('chibishibe')]).then((users) => {
```
I'd like to ask you to remove these changes. You are welcome to keep them local, but I'd like to preserve this and the `messages.json` file on my own repository.

Grissess replied:
Would it be desirable to merge all contributors so far, or just keep it at your version?
Also, even if it is left at just the two of you, could we do some fancy map/reduce to make it so there's not a variable per contributor? :)

imjoshdean commented on src/behaviors/shake/array/Alphaarray.js:
```diff
@@ -0,0 +1,2 @@
+var letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"].map(alpha => ":sh-" + alpha + ":");
```
This file can be deleted, as it is not used.

imjoshdean commented on src/behaviors/shake/shake.js:
```diff
@@ -0,0 +1,50 @@
+
+  parseMessage(message, channel, data) {
+   let splitMessage = message.replace(/^!shake/gi, '')
+   		.replace(/\:\w*\:/gi, '')
```
Let's bring down the indentation a bit, only needs to be two spaces from the let. 👌

imjoshdean commented on src/behaviors/shake/shake.js:
```diff
@@ -0,0 +1,50 @@
+		.replace(/[^a-zA-Z ]/g, '')
+		.trim()
+		.split('');
+	  if( splitMessage.length > 20) {
```
Trim the spacing before `splitMessage`

imjoshdean commented on src/behaviors/shake/shake.js:
```diff
@@ -0,0 +1,50 @@
+	  }
+
+const parsedMessage = splitMessage.map(alpha => alpha == " " ? ":ws:" : `:sh-${alpha}:`);
+
```
Unnecessary whitespace. In general this file has some wonky spacing on it. Try and clean it up, tab === two spaces please.

imjoshdean replied:
Other things I'm noticing, please use single quotes, not double quotes throughout.

imjoshdean commented on src/index.js:
```diff
@@ -23,8 +31,8 @@ const mascot = new MascotBot({
       settings: {
-        conDate: '8/11/2017',
-        maxDays: 398,
+        conDate: '2018-07-26 09-05:00',
```
This isn't a valid date. Maybe stick to a simpler format.

Grissess replied:
Our local test disagrees '-'
That's _not quite_  ISO 8601, but I believe the library which supports parsing those to times was complaining that the _previous_ version was going to be deprecated, so we saw to it to change it to this form. (It's missing MM:SS and T, isn't it :)
ISO 8601 is pretty unambiguous, would you prefer it in proper form instead?

imjoshdean replied:
Sorry, I didn't realize that was ISO-8601, as long as moment has no problems with it!

chibishibe replied:
Yeah, when I went to run the countdown behaviour for the first time. I encountered a Depreciation Warning in console. It worked, it just threw a warning out, so I tweaked the format a bit in order to keep a clean console

imjoshdean reviewed (changes requested):
A lot of non-difficult changes. Looking good, just needs polish!

chibishibe committed:
Went through recommended changed from recent git pull request.

clap -> Changed to emp, added functionality
days until -> Changed errors in indent
eventbrite - > Added correct amount of periods for ellipsis personality
greet -> changed to lowercase for icon
introductions -> reverted all changed to intoductions and messages
messages -> see above
Removed rave structure -> combined into changed emp
Removed array -> No longer needed
shake.js -> Fixed indenting and removed whitespace
indec -> Removed rave entry, added emp

chibishibe committed:
Launch Error, immediate fix

Index.js -> Forgot to remove clap.js changed to emp.js

imjoshdean commented on src/behaviors/emp/emp.js:
```diff
@@ -1,22 +1,22 @@
 import Behavior from '../behavior.js';
 
-class ClaporRave extends Behavior {
+class Emp extends Behavior {
```
Please use meaningful names and try to refrain from confusing abbreviations or shortened versions of words. While you might understand what they mean, other developers may not. `Emphasis` would be easier to understand.

imjoshdean commented on src/behaviors/shake/shake.js:
```diff
@@ -1,48 +1,46 @@
-const parsedMessage = splitMessage.map(alpha => alpha == " " ? ":ws:" : `:sh-${alpha}:`);
-
+    let splitMessage = message.replace(/^!shake/gi, '')
+    .replace(/\:\w*\:/gi, '')
```
This one is my bad for not being the clearest, you should definitely indent, but generally it's one tab off the current indentation so rather than

```js
let splitMessage = message.replace(/^!shake/gi, '')
.replace(/\:\w*\:/gi, '')
```

```js
let splitMessage = message.replace(/^!shake/gi, '')
  .replace(/\:\w*\:/gi, '')
``

imjoshdean reviewed (changes requested):
Still a lot of changes suggested not yet implemented, and a few more from here.

chibishibe committed:
-/+ log, (emp)(indentShake)(dependencies)

Removed non-current dependencies from package. ((airtable + eslint))
Changed name of "emp" to "emphasis" to improve clarity
Fixed indenting on 1:1 scale with shake
replaced changes in index ((Emp -> emphasis))

chibishibe committed:
This is a test of GitCola

GitCola will be my GitGui on Raspberri Pi, I need to test a commit

-Tweak response to greet.js

Grissess committed:
Post help in only one message

chibishibe committed:
Merge pull request #2 from chibishibe/single_help_msg

Post help in only one message

Tested, working correctly. Merging
